Update User Guide - Map views plugin (#8914)
geosolutions-it/MapStore2: docs/user-guide/map-views.md

# Map Views

> The **Map Views** is a [MapStore](https://mapstore.geosolutionsgroup.com/mapstore/#/) tool useful to configure multiple 3D views and switch between them with navigation tool.

Reviewer: It works also in 2D even if it has less options, I don't know if we want mention this

<img src="../img/map-views/add-view.gif" class="ms-docimage"/>

[MapStore](https://mapstore.geosolutionsgroup.com/mapstore/#/) allows to customize and edit the new view by opening the Edit panel through the <img src="../img/button/editing-button.jpg" class="ms-docbutton" /> button. Here the user is allowed to:

* Add a text, images, videos or hyperlinks by clicking **Description** section. The *description* is visible during the [Views Navigation](map-views.md#3d-views-navigations). Take a look at the following example.

<img src="../img/map-views/description.jpg" class="ms-docimage"/>

* Change the longitude, latitude and height of the *Camera Position* and *Center Position* by clicking on **Position** section.

<img src="../img/map-views/position.jpg" class="ms-docimage"/>

* Modify the duration of the animation and enable/disable the transition during the [Views Navigation](map-views.md#3d-views-navigations) by clicking on **Animation** section.

<img src="../img/map-views/animation.jpg" class="ms-docimage"/>

> * Select a WFS layer to create a mask of a 3D Tile layer by clicking **Mask** section. The WFS layer is added to the map by the user and then used in a transparent way to create the clipping area.

Reviewer: Maybe we should mention that this configuration has effect on all the 3D tiles visible in the map

> * Select a WFS layer to create a mask of a 3D Tile layer by clicking **Mask** section. The WFS layer is added to the map by the user and then used in a transparent way to create the clipping area.

Reviewer: ```suggestion
* Select a WFS or Vector layer available in the map to create a mask of a 3D Tile layer by clicking **Mask** section. The WFS or Vector layer is added to the map by the user and then used in a transparent way to create the clipping area.
```

<img src="../img/map-views/mask-panel.jpg" class="ms-docimage"/>

> If the user enables the **Inverse** option, a hole is created inside the 3D Tile layers. This hole can be filled by other 3D tiles or 3D objects.

Reviewer: ```suggestion
If the user enables the **Inverse** option, a hole is created inside the 3D Tile layers. This hole can be filled by only 3D objects (3D Tiles cannot be used to fill these holes because all 3D tiles will be masked).
```

<img src="../img/map-views/mask-inverse.jpg" class="ms-docimage"/>

If the user disables the **Inverse** option the clipping style is used for the view, then the selected WFS layer masks the 3D Tile layers outside its polygon.

<img src="../img/map-views/mask.jpg" class="ms-docimage"/>

!!! note

>     The WFS layer can be only polygon feature and concave polygon are not supported by this type of clipping, only convex layers can be used from this purpose.

Reviewer: This should refer to clipping inside the layer and not the mask option. So need to be moved below or merged with the clipping description of the layer

* Enable the translucency by clicking **Globe Translucency** section to view layers located below sea level.

<img src="../img/map-views/translucency.jpg" class="ms-docimage"/>

* Customize the layer present on the view by clicking **Layer Options** section. The user can simply enable or disable all the layers present on [TOC](toc.md#table-of-contents) and change the opacity of the layer or clipping for each 3D Tiles layer and are using another layer.

> For this operation, the user can select a layer from the *Clipping layer source* dropdown menu and choose a *Clipping feature* of the layer from the dropdown list where the hole on the 3D Tiles will be created.

Reviewer: *Clipping feature* is the polygon that should be convex anyway mapstore will try to transform it if it is concave

> For this operation, the user can select a layer from the *Clipping layer source* dropdown menu and choose a *Clipping feature* of the layer from the dropdown list where the hole on the 3D Tiles will be created.

Reviewer: ```suggestion
For this operation, the user can select a WFS or Vector layer available in the map from the *Clipping layer source* dropdown menu and choose a *Clipping feature* of the layer from the dropdown list where the hole on the 3D Tiles will be created.
```

!!! note

>     Unlike the *Mask* option described above, the *Clipping* area is visible only on the selected 3D Tiles layer and not on the whole view, and it is possible, using the *Clip function*, to select the feature of the WFS layer's attribute table that you want to use to do the clipping.

Reviewer: ```suggestion
    Unlike the *Mask* option described above, the *Clipping* area is visible only on the selected 3D Tiles layer and not on the whole view, and it is possible, using the *Clip function*, to select the feature of the WFS or Vector layer's attribute table that you want to use to do the clipping.
```

<img src="../img/map-views/clipping.gif" class="ms-docimage"/>

## 3D Views navigations

> Once a multiple views are added to the *Map Views* tool, [MapStore](https://mapstore.geosolutionsgroup.com/mapstore/#/) allows to visualize the view in sequence. By clicking on the <img src="../img/button/timeline-play-button.jpg" class="ms-docbutton" /> button, each view is displayed in the Map Viewer, with the previously set duration, and the description panel on the left side of the screen.

Reviewer: ```suggestion
Once multiple views are added to the *Map Views* tool, [MapStore](https://mapstore.geosolutionsgroup.com/mapstore/#/) allows to visualize them in sequence. By clicking on the <img src="../img/button/timeline-play-button.jpg" class="ms-docbutton" /> button, each view is displayed in the Map Viewer, with the previously set duration, and the description panel on the left side of the screen.
```

<img src="../img/map-views/map-views-panel.jpg" class="ms-docimage"/>

To create a new view, the user can, using the compass located in the upper right corner of the map, highlight a specific area on the map, enable the layers, on TOC, to be displayed in the map and finally click on the <img src="../img/button/+++.jpg" class="ms-docbutton" />. This view will be visible in the *Views* list by clicking the <img src="../img/button/timeline-layers-list-button.jpg" class="ms-docbutton" /> button.

<img src="../img/map-views/add-view.gif" class="ms-docimage"/>

[MapStore](https://mapstore.geosolutionsgroup.com/mapstore/#/) allows to customize and edit the new view by opening the Edit panel through the <img src="../img/button/editing-button.jpg" class="ms-docbutton" /> button. Here the user is allowed to:

* Add a text, images, videos or hyperlinks by clicking **Description** section. The *description* is visible during the [Views Navigation](map-views.md#3d-views-navigations). Take a look at the following example.

<img src="../img/map-views/description.jpg" class="ms-docimage"/>

> * Change the longitude, latitude and height of the *Camera Position* and *Center Position* by clicking on **Position** section.

Reviewer 2: In case of 3D ... otherwise?

<img src="../img/map-views/position.jpg" class="ms-docimage"/>

* Modify the duration of the animation and enable/disable the transition during the [Views Navigation](map-views.md#3d-views-navigations) by clicking on **Animation** section.

<img src="../img/map-views/animation.jpg" class="ms-docimage"/>

> * Select a WFS or Vector layer available in the map to create a mask on on all the 3D tiles visible in the map by clicking **Mask** section. The WFS or Vector layer is added to the map by the user and then used in a transparent way to create the clipping area.

Reviewer 2: Please correct the typo: "...on on all..."

It should be a bit well explained to explain how the clipping is realized using the WPS/vector layer (eg. the layers features are the areas to use for the mask...)

<img src="../img/map-views/mask-panel.jpg" class="ms-docimage"/>

> If the user enables the **Inverse** option, a hole is created inside the 3D Tile layers. This hole can be filled by only 3D objects (3D Tiles cannot be used to fill these holes because all 3D tiles will be masked).

Reviewer 2: "... This hole can be filled by only 3D objects (3D Tiles cannot be used to fill these holes because all 3D tiles will be masked)."

@allyoucanmap I don't think that sentence is correct. Can you confirm?

<img src="../img/map-views/mask-inverse.jpg" class="ms-docimage"/>

If the user disables the **Inverse** option the clipping style is used for the view, then the selected WFS layer masks the 3D Tile layers outside its polygon.

<img src="../img/map-views/mask.jpg" class="ms-docimage"/>

> * Enable the translucency by clicking **Globe Translucency** section to view layers located below sea level.

Reviewer 2: * Enable the translucency of the globe in **Globe Translucency** section so that it is possible to ...

not only under the sea level, it is called globe translucency indeed.

<img src="../img/map-views/translucency.jpg" class="ms-docimage"/>

> * Customize the layer present on the view by clicking **Layer Options** section. The user can simply enable or disable all the layers present on [TOC](toc.md#table-of-contents) and change the opacity of the layer or clipping for each 3D Tiles layer and are using another layer.

Reviewer 2: This must be well explained to also distinguish what is possible in 2D and what in 3D.

For this operation, the user can select a WFS or Vector layer available in the map from the *Clipping layer source* dropdown menu and choose a *Clipping feature* of the layer from the dropdown list where the hole on the 3D Tiles will be created.

<img src="../img/map-views/clipping.gif" class="ms-docimage"/>

!!! warning
    The WFS layer can be only polygon feature and concave polygon are not supported by this type of clipping, only convex layers can be used from this purpose.

!!! note

>     Unlike the *Mask* option described above, the *Clipping* area is visible only on the selected 3D Tiles layer and not on the whole view, and it is possible, using the *Clip function*, to select the feature of the WFS or Vector layer's attribute table that you want to use to do the clipping.

Reviewer 2: To many commas. Please well reformulate.

## 3D Views navigations

> Once multiple views are added to the *Map Views* tool, [MapStore](https://mapstore.geosolutionsgroup.com/mapstore/#/) allows to visualize them in sequence. By clicking on the <img src="../img/button/timeline-play-button.jpg" class="ms-docbutton" /> button, each view is displayed in the Map Viewer, with the previously set duration, and the description panel on the left side of the screen.

Reviewer 2: Once multiple views are added to the *Map Views* tool it allows to visualize them in sequence by clicking on the <img src="../img/button/timeline-play-button.jpg" class="ms-docbutton" /> button. Doing that the presentation mode starts and each view is displayed in the Map Viewer, together with its descriptive panel (if configured) on the left side of the screen, for a time depended on the duration previously configured.

<img src="../img/map-views/views-navigation.gif" class="ms-docimage"/>

> The user can also pause navigation or go back and forth in viewing views thanks to the navigation toolbar.

Reviewer 2: The user can also choose to navigate each view manually using the navigation toolbar provided by the tool.

* Add a text, images, videos or hyperlinks by clicking **Description** section. The *description* is visible during the [Views Navigation](map-views.md#3d-views-navigations). Take a look at the following example.

<img src="../img/map-views/description.jpg" class="ms-docimage"/>

* *Capture the view positions* through the <img src="../img/button/capture-view-position-button.jpg" class="ms-docbutton" /> button or change the longitude, latitude and height of the *Camera Position* and *Center Position*  (only available for the *3D* mode) by clicking on **Position** section.

<img src="../img/map-views/position.jpg" class="ms-docimage"/>

* Modify the duration of the animation and enable/disable the transition during the [Views Navigation](map-views.md#3d-views-navigations) by clicking on **Animation** section.

<img src="../img/map-views/animation.jpg" class="ms-docimage"/>

* Select a WFS or Vector layer available in the map to create a mask on all the 3D tiles visible in the map by clicking **Mask** section. The WFS or Vector layer is added to the map by the user and then used the layers features in a transparent way to create the clipping area.

> <img src="../img/map-views/mask-panel.jpg" class="ms-docimage"/>

Reviewer 2: Use also this image here 
https://github.com/geosolutions-it/MapStore2/pull/8914/files#diff-1899e3a95199c8d906cb177c7bc58aecd978c37497e402b5a7ad13024bffdd8aR42

> If the user enables the **Inverse** option, a hole is created inside the 3D Tile layers. This hole can be filled by only 3D objects (3D Tiles cannot be used to fill these holes because all 3D tiles will be masked).

Reviewer 2: If the user enables the **Inverse** option, simply the inverse mask is applied using the same layer so that each feature is used to produce an hole on all visible 3D Tiles.

<img src="../img/map-views/mask-inverse.jpg" class="ms-docimage"/>

> If the user disables the **Inverse** option the clipping style is used for the view, then the selected WFS layer masks the 3D Tile layers outside its polygon.

Reviewer 2: remove this line

> <img src="../img/map-views/mask.jpg" class="ms-docimage"/>

Reviewer 2: Remove this here, used above

> * Enable the translucency of the globe in **Globe Translucency** section so that it is possible to see layers beneath the globe's surface.

Reviewer 2: * Enable the translucency of the globe through **Globe Translucency** section so that it is possible to see layers under the globe's surface.

<img src="../img/map-views/translucency.jpg" class="ms-docimage"/>

> * Customize the layer present on the view by clicking **Layer Options** section. The user can simply enable or disable all the layers present on [TOC](toc.md#table-of-contents) and change the opacity of the layer.

Reviewer 2: * Choose which layer should be visible when the view is active through the **Layer Options** section. In 2D mode the user can simply enable or disable all the layers present on [TOC](toc.md#table-of-contents) and change the opacity.

<img src="../img/map-views/layer-options-tool.jpg" class="ms-docimage"/>

!!! warning

>     In the *2D mode*, the *3D Tiles* layer and the *Terrain* are not displayed in the *Layer Options*.

Reviewer 2: In *2D mode*, the *3D Tiles* and the *Terrain* layers are not displayed in the *Layer Options*.

> For the *3D mode*, it is also possible to clip each layer of 3D tiles using another layer. For this operation, the user can select a WFS or Vector layer available in the map from the *Clipping layer source* dropdown menu and choose a *Clipping feature* of the layer from the dropdown list where the hole on the 3D Tiles will be created.

Reviewer 2: In *3D mode*, using the same logic described above for the Masking option, it is also possible to Clip (not Mask) each 3D Tiles or Terrain layers using a WFS or a vector layer as a clipping source. Furthermore, in this case the user can also choose which layer feature can be used as *Clipping feature*.

<img src="../img/map-views/clipping.gif" class="ms-docimage"/>

!!! warning

>     The WFS layer can be only polygon feature and concave polygon are not supported by this type of clipping, only convex layers can be used from this purpose.

Reviewer 2: The clipping layer must have polygon convex features. Concave polygons are not supported by this type of clipping.

!!! note

>     Unlike the *Mask* option, described above, the *Clipping* area is a more narrowly focused tool because:

Reviewer 2: Unlike the *Mask* option, described above, the *Clip function* is a more narrowly focused tool because:

    * The clipping are is visible only on the selected 3D Tiles layer and not on the whole view

>     * It is possible, using the *Clip function*, to select the feature of the WFS or Vector layer's attribute table that you want to use to do the clipping

Reviewer 2: * It is possible to select the feature of the WFS or Vector layer that you want to use for the clipping
